inst/templates/ad_adae.R

```diff
@@ -110,6 +116,14 @@ adae <- ae %>%
     mode = "last"
   )
 
+# Join all ADSL with AE
+adae <- adae %>%
+
+  select(-TRTSDT, -TRTEDT, -DTHDT, -EOSDT) %>%
```

Comment:
@thomas-neitmann, @aehmann-gsk  : Should we implement something to avoid this? These adsl variables are added at the start of the pgm to help derivation of other variables and removed at the end before merging again with the full adsl. 
Ideally, we would define only once the variables to add/remove.
We have discussed this already, but not sure this is on the plan yet?

Reply:
For the moment, I have tried to drop the ADSL vars once they have been used. Then add all ADSL vars at the end, this avoids carrying through lots of variables, and having to update them for new records created (long byvar statements).

Reply:
How about something like this at the start:

```r
adsl_vars <- vars(TRTSDT, TRTEDT, DTHDT, EOSDT)
adae <- ae %>%
  left_join(select(adsl, !!!adsl_vars))
```

and at the end

```r
adae <- adae %>%
  left_join(select(adsl, -!!!adsl_vars))
```

Reply:
That's the same I had in mind.

For datasets with derived parameter we need to add `adsl_vars` to the `by_vars` parameter.

Reply:
I have implemented the function `negate_vars()` for it in the `546_derive_param_tte` branch. See https://github.com/Roche-GSK/admiral/blob/546_derive_param_tte/R/utils.R .

Reply:
Have updated all scripts creating VADs with regard to adding ADSL vars, as we discussed.